pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Message.java

```diff
@@ -208,6 +208,15 @@ public interface Message<T> {
      */
     byte[] getSchemaVersion();
 
+    /**
+     * Get schema ID of the message.
+     * PIP-420 provides a way to produce messages with external schema,
+     * and the schema ID will be set to the message metadata.
+     *
+     * @return schema ID of the message if the message is produced with external schema.
```

Comment:
Please note `null` value can be returned if it does not have a schema id.

I also noticed an empty array also represents "no schema id". Please make the behavior clear in documents.